### PR TITLE
APS-1331 - Add gender to approved premise entity

### PR DESCRIPTION
**Pull request by davidatkinsuk**
This also adds a new endpoint to list approved premises with an optional gender-specific filter

It also introduces an ApprovedPremisesRepository to better split CAS1 code away from other CASs

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas1/Cas1PremisesController.kt [hunk 1]
```diff
@@ -29,4 +32,20 @@ class Cas1PremisesController(
...
       )
   }
+
+  override fun getPremisesSummaries(gender: Cas1ApprovedPremisesGender?): ResponseEntity<List<Cas1PremisesBasicSummary>> {
+    return ResponseEntity
+      .ok()
+      .body(
+        cas1PremisesService.getPremises(
+          gender = when (gender) {
+            Cas1ApprovedPremisesGender.male -> ApprovedPremisesGender.MALE
```
Comment (Rosie-Brigham) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/cas1/Cas1PremisesController.kt:42]: I believe this needs to be diffrent, let me just check...

File: src/main/resources/static/codegen/built-cas1-api-spec.yml [hunk 2]
```diff
@@ -6558,4 +6608,9 @@ components:
         staffCode:
           type: string
       required:
-        - staffCode
\ No newline at end of file
+        - staffCode
+    Cas1ApprovedPremisesGender:
+      type: string
+      enum:
+        - male
```
Comment (Rosie-Brigham) [inline src/main/resources/static/codegen/built-cas1-api-spec.yml:6615]: See above comment

**Review by Rosie-Brigham — DISMISSED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/PremisesEntity.kt [hunk 3]
```diff
@@ -259,6 +267,11 @@ class ApprovedPremisesEntity(
   status,
 )
 
+enum class ApprovedPremisesGender {
+  MALE,
```
[the remark this answers is not in the document]
Comment (davidatkinsuk) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/PremisesEntity.kt]: Done. I did this on an additional commit as it was going to be too difficult to unpick the individual commit.

**Review by RobBoothMOJ — APPROVED** [review]
Approved

**Pull request merged**